src/Microsoft.Azure.WebJobs.Host/Tables/TableAttributeBindingProvider.cs

```diff
@@ -365,5 +425,76 @@ public ResolvedTableAttribute(TableAttribute inner, IStorageTableClient client)
 
             internal IStorageTableClient Client { get; private set; }
         }
+
+        // IStorageTable --> IQueryable<T>
+        // ConverterManager's pattern matcher will figure out TElement. 
+        private class Table2IQueryableConverter<TElement> where TElement : ITableEntity, new()
+        {
+            public Table2IQueryableConverter()
+            {
+                // We're now commited to an IQueryable. Verify other constraints. 
+                Type entityType = typeof(TElement);
+
+                if (!TableClient.ImplementsITableEntity(entityType))
+                {
+                    throw new InvalidOperationException("IQueryable is only supported on types that implement ITableEntity.");
+                }
+
+                TableClient.VerifyDefaultConstructor(entityType);
+            }
+
+            public IQueryable<TElement> Convert(IStorageTable value)
+            {
+                // If Table does not exist, treat it like have zero rows. 
+                // This means return an non-null but empty enumerable.
+                // SDK doesn't do that, so we need to explicitly check. 
+                Task<bool> t = Task.Run(() => value.ExistsAsync(CancellationToken.None));
```

Comment:
Majority are sync. However, with the pattern matching, we can push this up and allow both sync or async. 

---
In reply to: [96492103](https://github.com/Azure/azure-webjobs-sdk/pull/967#discussion_r96492103) [](ancestors = 96492103)